helix-rest/src/main/java/org/apache/helix/rest/server/resources/metadatastore/MetadataStoreDirectoryAccessor.java

```diff
@@ -124,7 +148,34 @@ public Response getShardingKeys(@QueryParam("realm") String realm) {
     return JSONRepresentation(responseMap);
   }
 
-  private void getHelixNamespace() {
+  @PUT
+  @Path("/metadata-store-realms/{realm}/sharding-keys/{sharding-key: .+}")
```

Comment:
Why did we decide to make this a PathParam instead of a QueryParam? I think QueryParam would be easier to use and would minimize unnecessary confusion.

Reply:
I actually take this back. I think this approach makes sense - the fact that sharding key strings contain "/"s threw me off. Resolving.